pkg/pool-manager/pool_test.go

```diff
@@ -414,4 +416,61 @@ var _ = Describe("Pool", func() {
+				BeforeEach(func() {
+					poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:02", &vmConfigMap)
+					Expect(poolManager).ToNot(Equal(nil), "should create pool-manager")
+				})
+
+				table.DescribeTable("should allocate mac-address correspond to the one specified in the networks annotation",
+					func(networkRequestAnnotation string) {
+						pod.Annotations = map[string]string{networksAnnotation: fmt.Sprintf("%s", networkRequestAnnotation)}
+
+						By("Request specific mac-address by adding the address to the networks pod annotation")
+						err := poolManager.AllocatePodMac(&pod)
+						Expect(err).ToNot(HaveOccurred(), "should allocate mac address and ip address correspond to networks annotation")
+
+						By("Convert obtained networks annotation JSON to multus.NetworkSelectionElement array")
+						obtainedNetworksAnnotationJson := pod.Annotations[networksAnnotation]
+						obtainedNetworksAnnotation := []multus.NetworkSelectionElement{}
+						err = json.Unmarshal([]byte(obtainedNetworksAnnotationJson), &obtainedNetworksAnnotation)
+						Expect(err).ToNot(HaveOccurred(), "should convert obtained annotation as json to multus.NetworkSelectionElement")
+
+						By("Convert expected networks annotation JSON to multus.NetworkSelectionElement array")
+						expectedNetworksAnnotation := []multus.NetworkSelectionElement{}
+						err = json.Unmarshal([]byte(networkRequestAnnotation), &expectedNetworksAnnotation)
+						Expect(err).ToNot(HaveOccurred(), "should convert expected annotation as json to multus.NetworkSelectionElement")
+
+						By("Compare between each obtained and expected network request according to the mac-address")
+						for _, obtainedNetwork := range obtainedNetworksAnnotation {
+							for _, expectedNetwork := range expectedNetworksAnnotation {
+								if obtainedNetwork.MacRequest == expectedNetwork.MacRequest {
+									Expect(obtainedNetwork).To(Equal(expectedNetwork), "test pod annotation should be equal to expected annotation")
+								}
+							}
+						}
+					},
+					table.Entry("with single ip-address request as string array",
+						`[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1"],"mac":"02:00:00:00:00:00"}]`),
+					table.Entry("with multiple ip-address request as string array",
+						`[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1","10.10.0.2","10.0.0.3"],"mac":"02:00:00:00:00:00"}]`),
+				)
+
+				table.DescribeTable("should fail to allocate requested mac-address", func(networkRequestAnnotation string) {
+					pod.Annotations = map[string]string{networksAnnotation: networkRequestAnnotation}
+					By("Request specific mac-address by adding the address to the networks pod annotation")
+					err := poolManager.AllocatePodMac(&pod)
+					Expect(err).To(HaveOccurred(), "should fail to allocate mac address due to bad annotation format")
+				},
+					table.Entry("with ip-address request as string instead of string array",
```

Comment:
No need for a describe table if there is only a single line. It extends the complexity and brings no value.

Reply:
I use DescribeTable here so I wont need to have the same exact tests with only one line change,where this line is actually the input for the test.
I added another case where we have multiple network requests, its much easier to do it like that.
Also you have the test logic is in once place so its easier to understand
what you think?